sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -608,6 +667,20 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
         val tz = ctx.addReferenceObj("timeZone", timeZone)
         (c, evPrim, evNull) => s"""$evPrim = UTF8String.fromString(
           org.apache.spark.sql.catalyst.util.DateTimeUtils.timestampToString($c, $tz));"""
+      case ar: ArrayType =>
+        (c, evPrim, evNull) => {
+          val bufferTerm = ctx.freshName("bufferTerm")
+          val bufferClass = classOf[StringBuffer].getName
+          val writeArrayToBuffer = codegenWriteArrayToBuffer(ar, ctx)
+          s"""
+             |$bufferClass $bufferTerm = new $bufferClass();
+             |if (!$evNull) {
```

Comment:
looking at the document of `CastFunction`, the `evNull` is actually `resultIsNull`, it's not about input but about result, so we should not check it here.

Reply:
ok.